engine/src/main/java/org/hibernate/validator/internal/engine/ValidatorImpl.java

```diff
@@ -1277,12 +1278,20 @@ else if ( propertyPathNode.getKey() != null ) {
 	}
 
 	/**
-	 * Must be called and stored for the duration of the stack call
-	 * A new instance is returned each time
+	 * Potentially wrap the {@link TraversableResolver} into a caching one.
+	 * <p>
+	 * If {@code traversableResolver} is {@code TraverseAllTraversableResolver.INSTANCE}, we don't wrap it and it is
+	 * returned directly.
+	 * <p>
+	 * If it is not, we wrap the resolver for caching. In this case, a new instance is returned each time and it should
+	 * be used only for the duration of a validation call.
 	 *
 	 * @return The resolver for the duration of a full validation.
 	 */
-	private TraversableResolver getCachingTraversableResolver() {
+	private static TraversableResolver wrapTraversableResolverForCachingIfRequired(TraversableResolver traversableResolver) {
+		if ( traversableResolver == TraverseAllTraversableResolver.INSTANCE ) {
```

Comment:
this doesn't help where one registers a custom traversable resolver to use, unfortunately -- so does not help my use case at all :(

any way we can either make wrapping configurable, or allow a user to opt-out of the caching behavior?

Reply:
Sure, it does not help if you're using a custom traversable resolver. Out of curiosity, what does your custom traversable resolver do?

I'll see what I can do about providing a way to disable the caching.

Reply:
I have effectively a feature toggle around a subset of configuration; if the feature that uses that configuration is not active, I don't cascade into that subset of configuration. Removing it would cause quite some user frustration :(

Perhaps something like:

```
        Validation.byDefaultProvider().configure()
            .traversableResolver(new CustomTraversableResolverThatIsReallyFast())
            .disableTraversableResolverCaching()
            .buildValidatorFactory().getValidator();
```

to ensure it is opt-in?

Appreciate your efforts on this project!

Reply:
Yes, that's what I had in mind. I'm gonna remove this commit from the PR and create another PR.

Reply:
@narbsy I created https://github.com/hibernate/hibernate-validator/pull/856